capstone-backend/src/main/java/com/google/sps/servlets/Utility.java

```diff
@@ -349,4 +352,116 @@ else if (key.equals("remove_"+name)) {
     }
     return getById(id, collectionReference, response, genericClass).get(0);
   }
-}
\ No newline at end of file
+
+  /**
+   * This method assures that the given post request is valid, contains the necessary fields, and 
+   * does not contain keys that do not match object fields exactly.
+   * @param {jsonObject} the supplied request body in json form
+   * @param {response} response returned from the call
+   * @param {fields} the Fields (contains name and type) of the class being used  
+   * @param {requiredFields} a list of the names of fields deemed "necessary" for creation of an object.
+   * @return boolean true if it is a valid post request, false if not
+   */
+  public static <T extends BaseEntity> boolean postErrorHandler(JsonObject jsonObject, HttpServletResponse response, 
+      Field[] fields, List<String> requiredFields) throws IOException {
+    List<String> list = new ArrayList<>();
+    List<String> fieldNames = Arrays.asList(fields).stream().map(f -> f.getName()).collect(Collectors.toList());
+    list.addAll(jsonObject.keySet());
+    list.retainAll(fieldNames);
+    // Every jsonObject key must match a field name. If not, this check will fail. 
+    if (list.size() != jsonObject.keySet().size()) {
+      System.err.println("Error: Not all parameters in the request body are fields of the given class.");
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+      return false;
+    } 
+    list.clear();
+    list.addAll(jsonObject.keySet().stream()
```

Comment:
Is there a reason you're filtering out all empty keys from `jsonObject`?

Reply:
I filter out empty keys to make sure that the required fields are "actually there" and not just value-less keys in the json. For example, if an `ownerID` is deemed necessary for Club creation, then an empty String ownerID would not suffice and the post should fail in this case.

Reply:
Ah, thanks! I missed that you were filtering out keys with **values** that are empty. Can you add a quick update to the comment: "Every requiredField must be present as a key with a **non-empty** value..."?